lib/finder.rb

```diff
@@ -0,0 +1,26 @@
+class Finder
+
+  def initialize(ui, contacts)
+    @ui = ui
+    @contacts = contacts
+  end
+
+  def find_by(search_field)
+    keyword = ui.ask_for_search_keyword(search_field)
+    ui.display_all(search_result(search_field, keyword))
+  end
+
+  def search_result(search_field, keyword)
+    contacts_found = []
+    contacts.each do |contact|
```

Comment:
I believe there are more appropriate functions on Enum (like `#filter`) to filter a list by a certain criteria